### ISSUE TITLE
Unable to analyse use cases due to parsing error

### ISSUE THREAD

**Issue by mike0240** [NVISO-BE/ee-outliers#555]
Using the example use_cases provided, I was unable to run any of the example use cases due to the following error:

~~~
outliers_dev | 2020-07-13 07:44:36 - ERROR - error while analyzing use case
outliers_dev | Traceback (most recent call last):
outliers_dev |   File "outliers.py", line 276, in perform_analysis
outliers_dev |     analyzer.evaluate_model()
outliers_dev |   File "/app/analyzers/metrics.py", line 34, in evaluate_model
outliers_dev |     model_settings=self.model_settings)
outliers_dev |   File "/app/helpers/es.py", line 169, in count_and_scan_documents
outliers_dev |     total_events = self._count_documents(index, search_range, bool_clause, query_fields, search_query)
outliers_dev |   File "/app/helpers/es.py", line 149, in _count_documents
outliers_dev |     search_query=search_query))
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/client/utils.py", line 84, in _wrapped
outliers_dev |     return func(*args, params=params, **kwargs)
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/client/__init__.py", line 529, in count
outliers_dev |     "POST", _make_path(index, doc_type, "_count"), params=params, body=body
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/transport.py", line 358, in perform_request
outliers_dev |     timeout=timeout,
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py", line 261, in perform_request
outliers_dev |     self._raise_error(response.status, raw_data)
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/connection/base.py", line 182, in _raise_error
outliers_dev |     status_code, error_message, additional_info
outliers_dev | elasticsearch.exceptions.RequestError: RequestError(400, 'parsing_exception', 'Failed to parse')
outliers_dev | 2020-07-13 07:44:36 - ERROR - error while analyzing use case
outliers_dev | Traceback (most recent call last):
outliers_dev |   File "outliers.py", line 276, in perform_analysis
outliers_dev |     analyzer.evaluate_model()
outliers_dev |   File "/app/analyzers/word2vec.py", line 143, in evaluate_model
outliers_dev |     model_settings=self.model_settings)
outliers_dev |   File "/app/helpers/es.py", line 169, in count_and_scan_documents
outliers_dev |     total_events = self._count_documents(index, search_range, bool_clause, query_fields, search_query)
outliers_dev |   File "/app/helpers/es.py", line 149, in _count_documents
outliers_dev |     search_query=search_query))
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/client/utils.py", line 84, in _wrapped
outliers_dev |     return func(*args, params=params, **kwargs)
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/client/__init__.py", line 529, in count
outliers_dev |     "POST", _make_path(index, doc_type, "_count"), params=params, body=body
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/transport.py", line 358, in perform_request
outliers_dev |     timeout=timeout,
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py", line 261, in perform_request
outliers_dev |     self._raise_error(response.status, raw_data)
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/connection/base.py", line 182, in _raise_error
outliers_dev |     status_code, error_message, additional_info
outliers_dev | elasticsearch.exceptions.RequestError: RequestError(400, 'parsing_exception', 'Failed to parse')
outliers_dev | 2020-07-13 07:44:36 - ERROR - error while analyzing use case
outliers_dev | Traceback (most recent call last):
outliers_dev |   File "outliers.py", line 276, in perform_analysis
outliers_dev |     analyzer.evaluate_model()
outliers_dev |   File "/app/analyzers/terms.py", line 19, in evaluate_model
outliers_dev |     model_settings=self.model_settings)
outliers_dev |   File "/app/helpers/es.py", line 169, in count_and_scan_documents
outliers_dev |     total_events = self._count_documents(index, search_range, bool_clause, query_fields, search_query)
outliers_dev |   File "/app/helpers/es.py", line 149, in _count_documents
outliers_dev |     search_query=search_query))
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/client/utils.py", line 84, in _wrapped
outliers_dev |     return func(*args, params=params, **kwargs)
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/client/__init__.py", line 529, in count
outliers_dev |     "POST", _make_path(index, doc_type, "_count"), params=params, body=body
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/transport.py", line 358, in perform_request
outliers_dev |     timeout=timeout,
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py", line 261, in perform_request
outliers_dev |     self._raise_error(response.status, raw_data)
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/connection/base.py", line 182, in _raise_error
outliers_dev |     status_code, error_message, additional_info
outliers_dev | elasticsearch.exceptions.RequestError: RequestError(400, 'parsing_exception', 'Failed to parse')
outliers_dev | 2020-07-13 07:44:36 - ERROR - error while analyzing use case
outliers_dev | Traceback (most recent call last):
outliers_dev |   File "outliers.py", line 276, in perform_analysis
outliers_dev |     analyzer.evaluate_model()
outliers_dev |   File "/app/analyzers/terms.py", line 19, in evaluate_model
outliers_dev |     model_settings=self.model_settings)
outliers_dev |   File "/app/helpers/es.py", line 169, in count_and_scan_documents
outliers_dev |     total_events = self._count_documents(index, search_range, bool_clause, query_fields, search_query)
outliers_dev |   File "/app/helpers/es.py", line 149, in _count_documents
outliers_dev |     search_query=search_query))
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/client/utils.py", line 84, in _wrapped
outliers_dev |     return func(*args, params=params, **kwargs)
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/client/__init__.py", line 529, in count
outliers_dev |     "POST", _make_path(index, doc_type, "_count"), params=params, body=body
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/transport.py", line 358, in perform_request
outliers_dev |     timeout=timeout,
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py", line 261, in perform_request
outliers_dev |     self._raise_error(response.status, raw_data)
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/connection/base.py", line 182, in _raise_error
outliers_dev |     status_code, error_message, additional_info
outliers_dev | elasticsearch.exceptions.RequestError: RequestError(400, 'parsing_exception', 'Failed to parse')
outliers_dev | 2020-07-13 07:44:36 - ERROR - error while analyzing use case
outliers_dev | Traceback (most recent call last):
outliers_dev |   File "outliers.py", line 276, in perform_analysis
outliers_dev |     analyzer.evaluate_model()
outliers_dev |   File "/app/analyzers/simplequery.py", line 59, in evaluate_model
outliers_dev |     model_settings=self.model_settings)
outliers_dev |   File "/app/helpers/es.py", line 169, in count_and_scan_documents
outliers_dev |     total_events = self._count_documents(index, search_range, bool_clause, query_fields, search_query)
outliers_dev |   File "/app/helpers/es.py", line 149, in _count_documents
outliers_dev |     search_query=search_query))
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/client/utils.py", line 84, in _wrapped
outliers_dev |     return func(*args, params=params, **kwargs)
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/client/__init__.py", line 529, in count
outliers_dev |     "POST", _make_path(index, doc_type, "_count"), params=params, body=body
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/transport.py", line 358, in perform_request
outliers_dev |     timeout=timeout,
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/connection/http_urllib3.py", line 261, in perform_request
outliers_dev |     self._raise_error(response.status, raw_data)
outliers_dev |   File "/usr/local/lib/python3.6/site-packages/elasticsearch/connection/base.py", line 182, in _raise_error
outliers_dev |     status_code, error_message, additional_info
outliers_dev | elasticsearch.exceptions.RequestError: RequestError(400, 'parsing_exception', 'Failed to parse')
~~~

When I looked at the http data sent between ee-outliers and my elasticstack, I found this error:
~~~
POST /winlogbeat-*/_count HTTP/1.1
Host: localhost:9200
Accept-Encoding: identity
Content-Length: 212
connection: keep-alive
authorization: Basic [REDACTED]
content-type: application/json
user-agent: elasticsearch-py/7.5.1 (Python 3.6.11)

{"query":{"bool":{"filter":[{"range":{"\"@timestamp\"":{"gte":"2020-07-05T19:44:36.969584","lte":"2020-07-13T07:44:36.969605"}}},[{"query_string":{"query":"tags:endpoint AND _exists_:OsqueryFilter.cmdline"}}]]}}}

HTTP/1.1 400 Bad Request
content-type: application/json; charset=UTF-8
content-length: 388

{"error":{"root_cause":[{"type":"parsing_exception","reason":"Failed to parse","line":1,"col":130}],"type":"parsing_exception","reason":"Failed to parse","line":1,"col":130,"caused_by":{"type":"x_content_parse_exception","reason":"[1:130] [bool] failed to parse field [filter]","caused_by":{"type":"illegal_state_exception","reason":"expected value but got [START_ARRAY]"}}},"status":400}
~~~

When I remove the square brackets surrounding `{"query_string": ...` and run it again in kibana console, the problem is fixed.

Is it an issue with my elasticstack/ee-outliers configuration, or is it a bug.

**Comment by maximilienroberti** [comment]
Hello @mike0240 ,

Thanks for raising the issue! 

It seems that the problem is related to the Elasticsearch version you are using.
I was able to reproduce your error while passing from 7.6.2 to 7.8.0.

Which version are you using?

We are currently working on a solution to make it compatible with 7.8.0.
You can follow progress on this ticket!

**Comment by mike0240** [comment]
Ah, I was using Elasticsearch version 7.8.0. 

Thanks for the clarification.

**Comment by maximilienroberti** [comment]
Issue resolved on PR #557.
